Sources/SwiftDocC/Semantics/Metadata/Metadata.swift

```diff
@@ -192,5 +192,55 @@ public final class Metadata: Semantic, AutomaticDirectiveConvertible {
         
         return true
     }
+    
+    /// Validates the use of this Metadata directive in a documentation comment.
+    ///
+    /// Some configuration options of Metadata are invalid in documentation comments. This function
+    /// emits warnings for illegal uses and sets their values to `nil`.
+    func validateForUseInDocumentationComment(
+        symbolSource: URL?,
+        for bundle: DocumentationBundle,
+        in context: DocumentationContext,
+        problems: inout [Problem]
+    ) {
+        let invalidDirectives: [(any AutomaticDirectiveConvertible)?] = [
+            documentationOptions,
+            technologyRoot,
+            displayName,
+            callToAction,
+            pageKind,
+            _pageColor,
+            titleHeading,
+        ] + (redirects ?? [])
+          + supportedLanguages
+          + pageImages
+        
+        let namesAndRanges = invalidDirectives
+            .compactMap { $0 }
+            .map { (type(of: $0).directiveName, $0.originalMarkup.range) }
```

Comment:
I opted for iterating twice just to make the code a bit nicer so that we don't have to unwrap `$0` in the `map`. Although this could be written as:

```
.map { invalidDirective in invalidDirective.map { (type(of: $0).directiveName, $0.originalMarkup.range) } }
```

Reply:
Yes, I meant to use `compactMap` above. The code is equivalent, but the compactMap then map IMO is more readable/fluent but I don't have a strong opinion.